### PR TITLE
docs(contributing): update per recent changes

### PR DESCRIPTION
**Pull request by jackfranklin**
A few tidy ups and updates where I noticed some info was lacking / outdated.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -63,16 +63,27 @@ information on using pull requests.
 
 ## Code Style
 
-- Coding style is fully defined in [.eslintrc](https://github.com/puppeteer/puppeteer/blob/master/.eslintrc.js)
-- Code should be annotated with [closure annotations](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler).
-- Comments should be generally avoided. If the code would not be understood without comments, consider re-writing the code to make it self-explanatory.
+- Coding style is fully defined in [.eslintrc](https://github.com/puppeteer/puppeteer/blob/master/.eslintrc.js).
```
Comment (mathiasbynens) [inline CONTRIBUTING.md]:
```suggestion
- Coding style is fully defined in [`.eslintrc`](https://github.com/puppeteer/puppeteer/blob/master/.eslintrc.js).
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -158,19 +169,6 @@ Despite being named 'unit', these are integration tests, making sure public API
 npm run unit
 ```
 
-- To run tests in parallel, use `-j` flag:
```
Comment (mathiasbynens) [inline CONTRIBUTING.md:161]: Is this not useful anymore?

**Review by mathiasbynens — APPROVED** [review]
LGTM with a question

File: CONTRIBUTING.md (hunk 2 above)
Comment (jackfranklin) [reply CONTRIBUTING.md:161]: No, Mocha doesn't support this out of the box.

**Pull request merged**